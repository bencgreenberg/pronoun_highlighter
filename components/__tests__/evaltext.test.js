import React from 'react';
import Evaltext from '../evaltext';
import renderer from 'react-test-renderer';

it('renders gender female', () => {
  const tree = renderer
    .create(<Evaltext word="her" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders gender male', () => {
  const tree = renderer
    .create(<Evaltext word="him" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('render gender neutral', () => {
  const tree = renderer
    .create(<Evaltext word="them" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('does not match any list words', () => {
  const tree = renderer
    .create(<Evaltext word="sunset" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});