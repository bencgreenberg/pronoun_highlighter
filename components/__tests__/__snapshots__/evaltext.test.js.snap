// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not match any list words 1`] = `
<div
  className="jsx-920209421 textBox"
>
  <span
    className="jsx-920209421 neutralWord"
  >
    sunset
  </span>
</div>
`;

exports[`render gender female 1`] = `
<div
  className="jsx-920209421 textBox"
>
  <span
    className="jsx-920209421 femaleWord"
  >
    her
  </span>
</div>
`;

exports[`render gender neutral 1`] = `
<div
  className="jsx-920209421 textBox"
>
  <span
    className="jsx-920209421 neutralWord"
  >
    them
  </span>
</div>
`;

exports[`renders gender female 1`] = `
<div
  className="jsx-920209421 textBox"
>
  <span
    className="jsx-920209421 femaleWord"
  >
    her
  </span>
</div>
`;

exports[`renders gender male 1`] = `
<div
  className="jsx-920209421 textBox"
>
  <span
    className="jsx-920209421 maleWord"
  >
    him
  </span>
</div>
`;
